workspaces/src/network/builder.rs

```diff
@@ -0,0 +1,81 @@
+use std::future::{Future, IntoFuture};
+use std::marker::PhantomData;
+use std::path::PathBuf;
+
+use crate::network::Sandbox;
+use crate::{Network, Worker};
+
+pub type BoxFuture<'a, T> = std::pin::Pin<Box<dyn Future<Output = T> + Send + 'a>>;
```

Comment:
Interesting. Feels a little strange to do an additional allocation here, but I am guessing you can't just use `impl Future<...>` because the type is not concrete?

Reply:
it's because `impl Future` in a type alias is unstable 😞